sys-block/sg3_utils/sg3_utils-1.42.recipe

```diff
@@ -0,0 +1,116 @@
+SUMMARY="Utilities for interacting with the sg SCSI interface"
+DESCRIPTION="The sg3_utils package contains utilities that send SCSI commands \
+to devices. As well as devices on transports traditionally associated with SCSI \
```

Comment:
> For the contents. The goal is that it fits neatly in Terminal in "pkgman search" output. For the whole line there is already the usual 80 char limint and no reason to restrict it further.

Thanks @pulkomandy for [your answer](https://github.com/haikuports/haikuports/pull/2010#issuecomment-354592303) (copied above).
Now I have another corner case question. Here, this line
```
to devices. As well as devices on transports traditionally associated with SCSI \
```
is 81 chars **_if_** we count the `\`. I believe the backslash itself should be counted in the 80 chars limit, so we would need to move the "SCSI" to the next line. Am I right?

@alaviss, aren't these questions funny? Well, in case you need to edit this, here's the copy paste:
```
to devices. As well as devices on transports traditionally associated with \
SCSI (e.g. Fibre Channel (FCP), Serial Attached SCSI (SAS) and the SCSI \
Parallel Interface(SPI)) many other devices use SCSI command sets. ATAPI \
cd/dvd drives and SATA disks that connect via a translation layer or a bridge \
device are examples of devices that use SCSI command sets."
```

Reply:
The `\` is to concatenate the previous content with the next line, which means it get deleted. Furthermore, we only forcefully enforce the rule for `SUMMARY` as it's content is in `pkgman search` output and we want that content to fit within Terminal neatly (as @pulkomandy said). 

For other parts, I believe it's purely cosmetic as 80-char is just the right amount for something to fit within Terminal, making edits and reviews easier for everyone. So I believe a [+-]1 char diff won't harm much. Still, I can change this if required

Reply:
Yep, I know that the only really useful limit is the 70 chars limit for the contents of the SUMARY.

And I do admit that the purpose of the 80 chars limit is much less important, so I agree that 80±1 is not going to make much difference.

So for me this is OK if you prefer to not count the `\` in the 80 chars.

You are making **_so_** many **_great_** contributions that I am sure that **_if_** we had to choose between you and this arbitrary limit we would **_all_** choose to **_keep you_** and throw away this 80 chars limit ;)

Reply:
Well... in Haiku itself we are strict about the 80 char limit. I find this very useful because usually my text editor is exactly 80 columns wide, and I can use the other half of my screen for other things (for example a Terminal where I compile things, or a second editor viewing another file).

So for me a 81 column line will look like this:

    DESCRIPTION="The sg3_utils package contains utilities that send SCSI commands \
    to devices. As well as devices on transports traditionally associated with SCSI 
    \
    more text here..."

If we wanted the limit to be higher, we would have set it at 81 or 85 or 100 or 120 columns. But we chose 80. And, I'm sorry but no, I would not accept a contributor, no matter how proficient, if he deliberately decides to ignore the coding style for the project. Having a coherent coding style may seem unimportant, but it keeps the code consistent and a lot easier to read and maintain. In the long term, contributors come and go, but the code stays.

Reply:
Don't worry, @pulkomandy, I'm sure @alaviss was going to switch to the strict 80 chars limit. He even told he would do so if needed.

Reply:
Yes, I'm just explaining why it is important as it is not necessarily obvious. This is why I wrote that I would not accept people who *deliberately* break the rules. Everyone is allowed to make mistakes and (hopefully) learn from them :)

Reply:
I'm sorry, I owe you my apologies. I wrote that I would be OK with 80±1 chars, but it was not true ;) I was secretly hoping that @alaviss would rewrap the text,. And if this did not happen I would have waited until another key contributor like you would come and say we really need to respect that limit.
So, sorry for my not-so-smart strategy, and thanks for your reaction :)

Reply:
For completeness: also notice how github added an horizontal scrollbar to the block of code above this discussion. They do this when code goes over 80 chars, and they also use a strict limit.

Reply:
@alaviss, please accept my apologies. I should at least have written that it would be better to rerwap the text. Instead, I chose pure flattery... :)
When you have some time, could you copy/paste 
Could you replace lines 3 to 7 of your recipe with the following text?
```
to devices. As well as devices on transports traditionally associated with \
SCSI (e.g. Fibre Channel (FCP), Serial Attached SCSI (SAS) and the SCSI \
Parallel Interface(SPI)) many other devices use SCSI command sets. ATAPI \
cd/dvd drives and SATA disks that connect via a translation layer or a bridge \
device are examples of devices that use SCSI command sets."
```
Thanks! And please stay with us! We **_need_** you!

Reply:
@fbrosson I don't remember saying I'm going to leave lol

Thanks for the heads up, @pulkomandy.

Reply:
Also, does the 4-space tab size setup applies to HaikuPorts? If it doesn't then I would have to rewrap a lot of lines in the future

Reply:
Cool! Thanks @alaviss!

For the tabs, I think the rule is that they count for 8 chars (which is the default in vi as we all know), **but** the 80 chars limit allows exceptions in PATCH, BUILD, INSTALL and TEST.
I also think that the rule should only be applied strictly for the variables.
The rationale, if I understand correctly, is that if splitting a line (to try to respect the 80 limit) in any function (i.e. PATCH, BUILD, INSTALL, TEST) makes the code less readable than if that line was not split, then it is more or less OK to have a few long lines.

Reply:
Sure, I was curious if parts of [Haiku's Coding Guidelines](https://www.haiku-os.org/development/coding-guidelines) applies here. Looks not. Thanks @fbrosson

Reply:
The 4 spaces per tab rule definitely applies here.